### DISCUSSION TITLE
Resume printing after power failure?

### DISCUSSION THREAD

**Discussion by spitfire** [Guilouz/Creality-Helper-Script-Wiki#450] [Features Request]
AFAIK both K1 and KE offer resuming of print after power failure. I know I should have a UPS, but I don't have one (next to the printer) yet. Recently I had a VERY brief (less than a second) power outage which interrupted my 14-hour print, just an hour before finish...ouch. Is there a way to use the alternative (as in non-creality) stuff (mainsail, guppy screen) and be able to recover from that? If not - can this be implemented?

**Comment by sTiKyt** [comment]
No response? Did you find a way?

**Reply by spitfire** [reply]
I got a UPS for the printer;)

**Reply by sTiKyt** [reply]
Nice, but doesn't solve software problem :(

**Comment by NexGen-3D-Printing** [comment]
If you need this feature, Klipper based printers are not for you, its not possible to even implement as the printer would first need real end-stop limit switches in order to restart, that's the first hurdle, the control board would need some large capacitor base onboard UPS like power to provide some sustained power in order to safely park its process, its just a nightmare to even get working, specially on Creality machines.

I recommend you either buy a used ex enterprise UPS, I use a large HP one for one of my machines and my networking gear, and I have another couple running through a large Bluetti portable powerpack that has UPS feature, this one is awesome and can run the printers for hours on its own.

**Reply by sTiKyt** [reply]
But I mean.. Stock firmware has it, no? At least I remember seeing something like that back when I had it with stock stuff

**Reply by NexGen-3D-Printing** [reply]
I don't know, I jail broke my printers straight away, never even attempted to use the Creality setup, but if you search on the Klipper discourse, I think you will find its either not supported at all, or its not very well supported, but I 100% can not see how Creality could have done this without having actually hardware end stop switches and large onboard capacitance to allow the Linux OS to park and save all processes, these machines use sensorless homing, the printers I design also use sensorless home, this is perfectly fine for setting a zero point in the beginning, but its not accurate, you will have up to 1 mm variance from run to run, this is why it is not used for Z axis homing, so its just not possible after power-loss to pick things back up, if you have hardware endstop switches, at the very lest it can re-home the XY axis and attempt to kick off.

Personally, the absolute best option is a UPS, but I assume when you say "Software Problem" is when Klipper chucks a hissy fit and the firmware shuts down? There is nothing that can be done here, its a pain in the backside, as a low flying Swallow at less than its maximum unladen airspeed will cause Klipper to have a spat and ruin your print job :)

I will admit, and is the only reason I have switched to using these K1's is they have really done a good job with Klipper, these machines are far more stable and reliable than I expected, I have all sorts of issues with Multi MCU over both USB and CAN with all of my DIY printers, so many issues with Z homing, probing and Klipper just having a hissy fit and shutting down.

**Comment by mannebk79** [comment]
yep, stock creality 1.3.3.46 has power loss resume, and it works quite surprisingly well.

I use it when I cant keep my hands out and make a yx jam and the printer looses its position, i just switch it off, and then power up and continue. no issues. happens almost in every print to me. cant keep my hands away :-)

just upgraded to gruppy screen, dang it, my most loved feature is gone, while the rest is really awesome, im missing that PLR absolutely, and while i keep mainsail and fluidd, i will go back to the creality touch screen!

**Reply by mannebk79** [reply]
What I do in the mean time, is measure on my print bed what layer i was currently printing, 

then I place a cardboard on top and let it home. (i make sure i always have the parts places to my homing location is inside a print, if possible. 

in orca i split the model at the same level and let the lower part be discharged. then newly sliced and upload the new file with just the not printed part in the gcode. 

when i restart printing, i z-offset the thickness of my cardboard plus some. works also, but is a lot more work!

**Reply by mannebk79** [reply]
i just found this: https://github.com/bigtreetech/KlipperPLR, will try with my rooted K1C runing Moonraker, Gruppy Screen, Mainsail and Fluidd....

**Reply by dogucan97** [reply]
You can actually use the Creality Cloud mobile app to do PLR even if you have Guppy Screen installed. Though I don't know if you can connect the app to the printer _after_ you install Mainsail/Guppy on it.
